syntax = "proto3";

package proto;

service DiscoveryService {
	/*
		Consente di comunicare con il
		microservizio passato come parametro.
	*/
	rpc discoveryLogin(DiscoveryLoginRequest) returns (DiscoveryLoginReply);
	
	
	/*
		Consente di registrare la porta relativa
		al microservizio passato come parametro.
	*/
	rpc put(PutRequest) returns (PutReply);

}


/*
	serviceName: Nome del microservizio di cui si
		     sta richiedendo l'informazione
	             relativa alla porta.
*/
message DiscoveryLoginRequest {
	string serviceName = 1;
	string username = 2;
}
/*
	serviceName: Nome del microservizio di cui si
		     è chiesto l'informazione relativa
		     alla porta.
	             
	port: 	     valore della porta su cui il microservizio
	      	     è in ascolto
*/
message DiscoveryLoginReply {
	string validUsername = 1;
}





message PutRequest {
	string serviceName = 1;
	string port = 2;
}


/*
	result:    risultato dell'esecuzione. Il metodo
		         ritorna TRUE se l'esecuzione è andata
		         a buon fine; altrimenti, ritorna FALSE.
	list_server: Lista dei discovery server.
*/
message PutReply {
	bool result = 1;
}